### PR TITLE
OSC features added: save/load to userPatchPath, param extended options doc messages

### PR DESCRIPTION
**Pull request by gitsmol**
- Added Save/load to userPatchPath
- Added OSC error message when loading invalid path
- Added OSC doc message for param extended options
- Updated OSC specifications html accordingly

### CODE REVIEW

File: src/surge-xt/osc/OpenSoundControl.cpp [hunk 1]
```diff
@@ -24,12 +24,14 @@
...
 #include "SurgeSynthProcessor.h"
 #include "SurgeStorage.h"
+#include <iostream>
 #include <sstream>
 #include <vector>
 #include <algorithm>
 #include <string>
 #include "UnitConversions.h"
 #include "Tunings.h"
+#include "filesystem/ghc-filesystem.h"
```
Comment (baconpaul) [inline src/surge-xt/osc/OpenSoundControl.cpp:34]: this is not the right way to import filesystem. Instead do

```
#include "import/filesystem.h"
```

which will load one of `std::filesystem` or `ghc::filesystem` and set the namespace `fs` to point.

This is why your windows builds (where we don't use GHC) failed.

Comment (gitsmol) [reply src/surge-xt/osc/OpenSoundControl.cpp:34]: Much appreciated, Paul. However, `#include "import/filesystem.h"` doesn't compile for me. I see SurgeStorage uses `#include "filesystem/import.h"` though, so probably that's the right header?

Comment (baconpaul) [reply src/surge-xt/osc/OpenSoundControl.cpp:34]: yes! sorry about that

Comment (gitsmol) [reply src/surge-xt/osc/OpenSoundControl.cpp]: Looks it still doesn't compile on windows. I'll investigate later, looks like I should convert Path to string before feeding to strncpy.

**Pull request merged**